server/src/events.mjs

```diff
@@ -80,12 +80,14 @@ function getRegisteredEventListener(userId, method) {
 
 // Remove mapping from event listener request method name from our map
 // Attempts to send events to this listener going forward will fail
-function deregisterEventListener(userId, oMsg) {
-  if ( ! eventListenerMap[userId] ) { return; }
-  delete eventListenerMap[userId][oMsg.method];
-  logger.debug(`Deregistered event listener for method: ${userId}:${oMsg.method}`);
+function deregisterEventListener(userId, oMsg, ws) {
+  if (!eventListenerMap[userId]) { return; }
+  
+  if ( eventListenerMap[userId] && eventListenerMap[userId][oMsg.method] && eventListenerMap[userId][oMsg.method].ws === ws ) {
```

Comment:
Should we log some kind of error if the ws values don't match? Would this be an unexpected error to somehow track down? Or... should we *not* compare ws values and nuke the listener record no matter what?

Reply:
Hmm. That brings up a scenario I don't know if we thought of. What if a user is listening to the same method across multiple connections. Why they would do this I don't know but I guess it's possible.

Considering the fact that we're still limiting to one listen per user per method I'd say we just nuke it no matter what. But maybe we want to put in the feature for multiple listens per user per connection?

Keaton, what'd be the difficulty on that? I'm assuming we'd just be setting [ws] as an array and some logic on top to add/remove the proper one.

Reply:
Per your request, I was able to set ws key to an array. I have renamed it to wsArr and we can now store multiple ws connections inside of the array.